flink-runtime/src/main/java/org/apache/flink/runtime/security/factories/DefaultSecurityContextFactory.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.security.factories;
+
+import org.apache.flink.runtime.security.SecurityConfiguration;
+import org.apache.flink.runtime.security.contexts.HadoopSecurityContext;
+import org.apache.flink.runtime.security.contexts.NoOpSecurityContext;
+import org.apache.flink.runtime.security.contexts.SecurityContext;
+
+import org.apache.hadoop.security.UserGroupInformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Default security context factory that instantiates {@link SecurityContext}
+ * based on installed modules, it would instantiate {@link HadoopSecurityContext} if
+ * a {@link HadoopModuleFactory} is included.
+ */
+public class DefaultSecurityContextFactory implements SecurityContextFactory {
+	private static final Logger LOG = LoggerFactory.getLogger(DefaultSecurityContextFactory.class);
+
+	@Override
+	public SecurityContext createContext(SecurityConfiguration securityConfig) {
```

Comment:
It seem like this could be split up into two factories, once for Hadoop and one for no-op. Take a look at `PipelineExecutorFactory`, where the factory has a method for checking if the factory is applicable or not. We could use the same pattern here instead of making the `createContext()` method serve two purposes.